### ISSUE TITLE
docs(book): ast-traversal examples drifted from test guardian and current API

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#937]
## Summary

`tests/book_ast_traversal_examples.rs` exists to keep the runnable examples in `big-code-analysis-book/src/library/ast-traversal.md` honest, but two book examples have drifted out of sync with both the test and the current API — and the test does not catch it, because the test was updated while the book was not.

## Location

- Test file (the designated guardian): `tests/book_ast_traversal_examples.rs:1-6` (module doc), `:120-145`, `:147-184`
- Drifted book examples: `big-code-analysis-book/src/library/ast-traversal.md:221-254` and `:273-296`
- API: `src/ast.rs:230-243` (`AstCfg` now has a required `language: String` field), `src/metrics/nom.rs:121` (`functions_sum` returns `u64`)

## Evidence

The test module doc promises a two-way mirror:

```rust
//! ... so doc rot is caught by `cargo test` instead
//! of by readers trying to copy-paste broken snippets. If you change
//! the book, mirror the change here; if a refactor breaks an example
//! here, fix both.
```

Two divergences exist where the test was kept current but the book was not:

1. **`AstCfg` missing the `language` field (hard compile break).**
   `AstCfg` gained a required `language: String` field (`src/ast.rs:237`, #654). The test constructs it correctly:

   ```rust
   let cfg = AstCfg {
       id: payload.id.clone(),
       language: "rust".to_owned(),   // present in the test
       comment: payload.comment,
       span: payload.span,
   };
   ```

   The book example does NOT (`ast-traversal.md:283-287`):

   ```rust
   let cfg = AstCfg {
       id: payload.id.clone(),
       comment: payload.comment,      // no `language` field
       span: payload.span,
   };
   ```

   This fence is `rust,no_run` (`ast-traversal.md:273`) — compiled by `mdbook test` / rustdoc — so it would fail to compile (`missing field `language``).

2. **`functions_sum()` compared against a float (stale type).**
   `functions_sum` returns `u64` (`src/metrics/nom.rs:121`). The test asserts `== 3`. The book asserts `== 3.0` (`ast-traversal.md:252`) — a leftover from the f64-to-u64 metric type flip. This fence is `rust,ignore` (`ast-traversal.md:221`), so it is never compiled and the staleness is silent.

## Expected Behavior

The book examples should compile and match the current API and the test assertions, since the test's stated purpose is to keep them honest. `mdbook test` (or an equivalent gate) should be wired so the `rust` / `no_run` book examples are compile-checked in CI; the `rust,ignore` examples that the test mirrors should at minimum be kept type-accurate.

## Actual Behavior

- The `no_run` `AstCfg` example will not compile against the current crate (missing `language` field), but nothing in CI compiles it.
- The `ignore` `functions_sum() == 3.0` example uses a stale float type and is never compiled.
- The test guardian gives a false sense of safety: it asserts the correct shapes, but the book examples a reader actually copies are broken, and the test cannot fail on that drift.

## Impact

Readers who copy-paste the book's `Ast::dump` (`AstCfg`) example get a compile error; readers who copy the metrics-plus-symbol-table example get a type-mismatch error. The module doc's "doc rot is caught by `cargo test`" claim is only partially true — it protects the test's copy of the code, not the book's.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

The book's `ast-traversal.md` examples were not updated in lockstep with two API changes that the mirroring test (`tests/book_ast_traversal_examples.rs`) did pick up: the `AstCfg.language` field (#654) and the `functions_sum()` u64 return. Because no CI step runs `mdbook test` over these fences (the pages workflow only does `mdbook build`), the `no_run` example's compile break and the `ignore` example's stale float are invisible.

### Implementation Steps

1. In `big-code-analysis-book/src/library/ast-traversal.md`:
   - Add `language: "rust".to_owned(),` to the `AstCfg` literal at lines 283-287 so the `rust,no_run` example matches `src/ast.rs:232-243`.
   - Change `assert_eq!(space.metrics.nom.functions_sum(), 3.0);` (line 252) to `== 3` to match the `u64` return.
2. Diff every other code fence in the file against `tests/book_ast_traversal_examples.rs` for any further drift while in there.
3. (Recommended, separate change) Wire `mdbook test` for this crate's book into CI so `rust` / `no_run` fences are compile-checked, closing the silent-rot gap that this issue exposes. Track alongside #917 (the analogous Python book-example gate gap).

### Tests to Add/Update

- No new Rust test needed; `tests/book_ast_traversal_examples.rs` already asserts the correct shapes.
- If `mdbook test` is wired (step 3), it becomes the regression gate for the book copies themselves.

### Verification

- [ ] `cargo test --test book_ast_traversal_examples --features rust` still passes
- [ ] `mdbook test big-code-analysis-book` compiles the `rust` / `no_run` fences (after step 1 fix)
- [ ] Manual: copy-paste the `Ast::dump` and metrics-plus-symbol-table examples into a scratch binary; both compile

**Comment by dekobon** [comment]
Fixed in `c4d2ad5b` (branch `worktree-agent-a6ca7d8b6a7203a09`, not yet pushed).

Both drifts in `big-code-analysis-book/src/library/ast-traversal.md` are corrected to match the guardian (`tests/book_ast_traversal_examples.rs`) and the current API:

- The `no_run` `AstCfg` literal now sets `language: "rust".to_owned()` (the field `AstCfg` gained in #654).
- The `ignore` example compares `functions_sum()` against `3` (`u64`), not the stale `3.0`.

Verification:
- `cargo test --test book_ast_traversal_examples --features rust` — all 6 pass.
- The fixed `no_run` fence body was extracted into a scratch crate against this checkout and compiles cleanly.
- `rumdl check` clean; `mdbook build` succeeds.

I audited every other code fence in the page against the guardian test; no further drift remained.

Note on step 3 of the plan (wiring `mdbook test` into CI): left out of this change. It is a CI/infra change, not a docs fix, and `mdbook test -L target/debug/deps` cannot resolve the hashed rlib by crate name in this workspace today, so wiring it needs more than a one-line job. Recommend tracking it as a separate issue alongside #917, as the plan suggests. Not closing this issue.
